app/views/customizable/_form.html.erb

```diff
@@ -27,13 +27,15 @@ See doc/COPYRIGHT.rdoc for more details.
 
 ++#%>
 
-<%# locals: form %>
+<%# locals: form, all_fields, only_required #%>
 
 <% form.object.custom_field_values.each do |value| %>
   <%= form.fields_for_custom_fields :custom_field_values, value do |value_form| %>
     <% required = value.custom_field.is_required? %>
     <div class="form--field <%= required ? '-required' : '' %>">
-      <%= value_form.custom_field %>
+      <% if all_fields || (only_required && required) || (!only_required && !required) %>
```

Comment:
Alright, that's better. My point was also that you can be more concise about the second condition.
See

```
2.2.1 :016 > combinations = [[false, false], [false, true], [true, false], [true, true]]
 => [[false, false], [false, true], [true, false], [true, true]] 
2.2.1 :017 > combinations.map { |only_required, required| (only_required && required) || (!only_required && !required) }
 => [true, false, false, true] 
2.2.1 :018 > combinations.map { |only_required, required| only_required == required }
 => [true, false, false, true] 
```

But yeah, that's not absolutely necessary. A comment about what this means in English would be good either way IMO. I.e. explaining briefly when we show a custom field.